src/game/logic/object/weaponset.cpp

```diff
@@ -0,0 +1,259 @@
+                throw CODE_06;
+            }
+        }
+    } else if (xfer->Get_Mode() == XFER_SAVE) {
+        Utf8String name;
+        BitFlags<WEAPONSET_COUNT> flags;
+
+        if (m_curWeaponTemplateSet != nullptr) {
+            const ThingTemplate *tmplate = m_curWeaponTemplateSet->Get_Type();
+
+            if (tmplate == nullptr) {
+                throw CODE_06;
+            }
+
+            name = tmplate->Get_Name();
+            flags = m_curWeaponTemplateSet->Get_Conditions();
+        }
+
+        xfer->xferAsciiString(&name);
+        flags.Xfer(xfer);
+    }
+
+    for (int i = 0; i < WEAPONSLOT_COUNT; i++) {
+        bool has_weapon = m_weapons[i] != nullptr;
+        xfer->xferBool(&has_weapon);
+
+        if (has_weapon) {
+            if (xfer->Get_Mode() == XFER_LOAD && !m_weapons[i]) {
+                WeaponTemplate *weapon = m_curWeaponTemplateSet->Get_Nth(static_cast<WeaponSlotType>(i));
+
+                if (weapon == nullptr) {
+                    captainslog_dbgassert(false, "xfer backwards compatibility code - old save file???");
+                    weapon = m_curWeaponTemplateSet->Get_Nth(WEAPONSLOT_PRIMARY);
+                }
+
+                m_weapons[i] = WeaponStore::Allocate_New_Weapon(weapon, static_cast<WeaponSlotType>(i));
+            }
+
+            m_weapons[i]->Xfer_Snapshot(xfer);
+        }
+    }
+
+    xfer->xferUser(&m_curWeapon, sizeof(m_curWeapon));
+    xfer->xferUser(&m_curWeaponLocked, sizeof(m_curWeaponLocked));
+    xfer->xferUnsignedInt(&m_filledWeaponSlotMask);
+    xfer->xferInt(&m_totalAntiMask);
+    xfer->xferBool(&m_damageWeapon);
+    xfer->xferBool(&m_damageWeapon);
+    m_damageTypes.Xfer(xfer);
+}
+
+BitFlags<MODELCONDITION_COUNT> WeaponSet::Get_Model_Condition_For_Weapon_Slot(
+    WeaponSlotType slot, WeaponSetConditionType condition)
+{
+    static ModelConditionFlagType Nothing[WEAPONSLOT_COUNT] = {
+        MODELCONDITION_INVALID, MODELCONDITION_INVALID, MODELCONDITION_INVALID
+    };
+    static ModelConditionFlagType Firing[WEAPONSLOT_COUNT] = {
+        MODELCONDITION_FIRING_A, MODELCONDITION_FIRING_B, MODELCONDITION_FIRING_C
+    };
+    static ModelConditionFlagType Betweening[WEAPONSLOT_COUNT] = {
+        MODELCONDITION_BETWEEN_FIRING_SHOTS_A, MODELCONDITION_BETWEEN_FIRING_SHOTS_B, MODELCONDITION_BETWEEN_FIRING_SHOTS_C
+    };
+    static ModelConditionFlagType Reloading[WEAPONSLOT_COUNT] = {
+        MODELCONDITION_RELOADING_A, MODELCONDITION_RELOADING_B, MODELCONDITION_RELOADING_C
+    };
+    static ModelConditionFlagType PreAttack[WEAPONSLOT_COUNT] = {
+        MODELCONDITION_PREATTACK_A, MODELCONDITION_PREATTACK_B, MODELCONDITION_PREATTACK_C
+    };
+    static ModelConditionFlagType Using[WEAPONSLOT_COUNT] = {
+        MODELCONDITION_USING_WEAPON_A, MODELCONDITION_USING_WEAPON_B, MODELCONDITION_USING_WEAPON_C
+    };
+    ModelConditionFlagType *Lookup[WSF_COUNT] = { Nothing, Firing, Betweening, Reloading, PreAttack };
+
```

Comment:
assert on WSF_COUNT == 5
assert on WEAPONSLOT_COUNT == 3

For robustness.
This would (silently) break if WSF_COUNT, WEAPONSLOT_COUNT was ever increased.

Reply:
fixed.